platform/security/sts/security-sts-realm/src/main/java/ddf/security/realm/sts/AbstractStsRealm.java

```diff
@@ -237,24 +244,32 @@ protected SecurityToken requestSecurityToken(Object authToken) {
      * @param securityToken The token being renewed.
      * @return security token (SAML assertion)
      */
-    protected SecurityToken renewSecurityToken(SecurityToken securityToken) {
+    protected SecurityToken renewSecurityToken(final SecurityToken securityToken) {
         SecurityToken token = null;
         String stsAddress = getAddress();
 
         try {
             LOGGER.debug("Renewing security token from STS at: {}.", stsAddress);
 
             if (securityToken != null) {
-                LOGGER.debug("Telling the STS to renew a security token on behalf of the auth token");
-                STSClient stsClient = configureStsClient();
-
-                stsClient.setWsdlLocation(stsAddress);
-                stsClient.setTokenType(getAssertionType());
-                stsClient.setKeyType(getKeyType());
-                stsClient.setKeySize(Integer.parseInt(getKeySize()));
-                stsClient.setAllowRenewing(true);
-                token = stsClient.renewSecurityToken(securityToken);
-                LOGGER.debug("Finished renewing security token.");
+                synchronized (securityToken.getToken()) {
+                    SecurityToken cachedToken = cache.getIfPresent(securityToken.getToken());
```

Comment:
I think you should change the cache builder to `expireAfterAccess` instead of write. Otherwise, you could get the token here at 0:59 and return it but it would be evicted just after returning it (when the lock is released). Unless the semantics you want are for it potentially to be evicted if the timing worked out that way.

_Note: I'm not sure that Guava's cache tries to lock on the key; it's possible it would evict it before you returned. But either case seems wrong to me._